spec/opentsdb_consumer_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+RSpec.describe OpenTSDBConsumer do
+  let(:host)         { 'myhost.com' }
+  let(:port)         { 4242 }
+  let(:metric_names) { %w(metric1 metric2) }
+
+  describe '.fetch' do
+    subject do
+      described_class.fetch(
+        host: host, port: port,
+        metric_names: metric_names,
+        metric_options: { aggregator: 'cubic', start: 'now', unknown_key: 'not here' }
+      )
+    end
+
+    before(:each) do
+      allow(OpenTSDBConsumer::Client).to receive(:new)
+      allow(OpenTSDBConsumer::Metric).to receive(:new)
+
+      query_double = instance_double(OpenTSDBConsumer::Query)
```

Comment:
For this particular example, I agree with you on the fact that I overmocked things (I changed that already).

For the general case though, I don't share the opinion that mocking should only be done when calling an "external service" such as OpenTSDB, Facebook, etc. I think we should always mock these external calls, but not only them.

This is almost a philosophical topic, so I'll abstain from going further :).